packages/edit-site/src/components/page-library/search-items.js

```diff
@@ -0,0 +1,171 @@
+/**
+ * External dependencies
+ */
+import removeAccents from 'remove-accents';
+import { noCase } from 'change-case';
+
+// Default search helpers.
+const defaultGetName = ( item ) => item.name || '';
+const defaultGetTitle = ( item ) => item.title;
+const defaultGetDescription = ( item ) => item.description || '';
+const defaultGetKeywords = ( item ) => item.keywords || [];
+const defaultHasCategory = () => false;
+
+/**
+ * Extracts words from an input string.
+ *
+ * @param {string} input The input string.
+ *
+ * @return {Array} Words, extracted from the input string.
+ */
+function extractWords( input = '' ) {
+	return noCase( input, {
+		splitRegexp: [
+			/([\p{Ll}\p{Lo}\p{N}])([\p{Lu}\p{Lt}])/gu, // One lowercase or digit, followed by one uppercase.
+			/([\p{Lu}\p{Lt}])([\p{Lu}\p{Lt}][\p{Ll}\p{Lo}])/gu, // One uppercase followed by one uppercase and one lowercase.
+		],
+		stripRegexp: /(\p{C}|\p{P}|\p{S})+/giu, // Anything that's not a punctuation, symbol or control/format character.
+	} )
+		.split( ' ' )
+		.filter( Boolean );
+}
+
+/**
+ * Sanitizes the search input string.
+ *
+ * @param {string} input The search input to normalize.
+ *
+ * @return {string} The normalized search input.
+ */
+function normalizeSearchInput( input = '' ) {
+	// Disregard diacritics.
+	//  Input: "média"
+	input = removeAccents( input );
+
+	// Accommodate leading slash, matching autocomplete expectations.
+	//  Input: "/media"
+	input = input.replace( /^\//, '' );
+
+	// Lowercase.
+	//  Input: "MEDIA"
+	input = input.toLowerCase();
+
+	return input;
+}
+
+/**
+ * Converts the search term into a list of normalized terms.
+ *
+ * @param {string} input The search term to normalize.
+ *
+ * @return {string[]} The normalized list of search terms.
+ */
+export const getNormalizedSearchTerms = ( input = '' ) => {
+	return extractWords( normalizeSearchInput( input ) );
+};
+
+const removeMatchingTerms = ( unmatchedTerms, unprocessedTerms ) => {
+	return unmatchedTerms.filter(
+		( term ) =>
+			! getNormalizedSearchTerms( unprocessedTerms ).some(
+				( unprocessedTerm ) => unprocessedTerm.includes( term )
+			)
+	);
+};
+
+/**
+ * Filters an item list given a search term.
+ *
+ * @param {Array}  items       Item list
+ * @param {string} searchInput Search input.
+ * @param {Object} config      Search Config.
+ *
+ * @return {Array} Filtered item list.
+ */
+export const searchItems = ( items = [], searchInput = '', config = {} ) => {
```

Comment:
It would have been nice to be able to reuse the util functions from the block editor however our filtering needs in the library ended up differing to the point it was easier to bring these in and update as required given the short time to 6.3. 

When the dust settles we might be able to find some common ground and extract a shared util.